idseq_dag/steps/generate_coverage_viz.py

```diff
@@ -0,0 +1,823 @@
+
+            # Calculate the coverage for the accession, based on the reads and contigs.
+            (coverage, coverage_bin_size) = PipelineStepGenerateCoverageViz.calculate_accession_coverage(
+                accession_id, accession_obj, contig_data, read_data, num_bins
+            )
+
+            # Calculate statistics for the accession.
+            accession_stats = PipelineStepGenerateCoverageViz.calculate_accession_stats(
+                accession_obj,
+                contig_data,
+                read_data
+            )
+
+            coverage_viz_data[accession_id] = {
+                "total_length": total_length,
+                "name": accession_obj["name"],
+                "hit_groups": hit_groups,
+                "coverage": coverage,
+                "coverage_bin_size": coverage_bin_size,
+                "max_aligned_length": accession_stats["max_aligned_length"],
+                "coverage_depth": _format_number(accession_stats["coverage_depth"]),
+                "coverage_breadth": _format_percent(accession_stats["coverage_breadth"]),
+                "avg_prop_mismatch": _format_percent(accession_stats["avg_prop_mismatch"]),
+            }
+
+        return coverage_viz_data
+
+    # Generate a summary dict that maps taxons to accession data.
+    @staticmethod
+    def generate_coverage_viz_summary_data(taxon_data, accession_data, coverage_viz_obj):
+        coverage_viz_summary_data = {}
+
+        for taxon, data in taxon_data.items():
+            coverage_viz_summary_data[taxon] = {
+                "best_accessions": list(map(lambda accession_id: {
+                    "id": accession_id,
+                    "name": accession_data[accession_id]["name"],
+                    "num_contigs": len(accession_data[accession_id]["contigs"]),
+                    "num_reads": len(accession_data[accession_id]["reads"]),
+                    "score": _format_number(accession_data[accession_id]["score"]),
+                    "coverage_depth": coverage_viz_obj[accession_id]["coverage_depth"]
+                }, data["accessions"])),
+                "num_accessions": data["num_total_accessions"]
+            }
+
+        return coverage_viz_summary_data
+
+    # Get a dict that maps valid contigs to their read count.
+    # A contig is valid if it is larger than min_contig_size.
+    @staticmethod
+    def get_valid_contigs_with_read_counts(contig_stats_json, min_contig_size):
+        valid_contigs_with_read_counts = {}
+        with open(contig_stats_json, 'r') as csj:
+            contig_read_counts = json.load(csj)
+
+            for contig in contig_read_counts:
+                if contig != "*" and contig_read_counts[contig] >= min_contig_size:
+                    valid_contigs_with_read_counts[contig] = contig_read_counts[contig]
+
+        return valid_contigs_with_read_counts
+
+    # Generate a dict that maps accessions to the reads and contigs that were assigned to them.
```

Comment:
I believe for the python convention you will put the documentation inside the function with ''' ''' around it.

Reply:
Done.